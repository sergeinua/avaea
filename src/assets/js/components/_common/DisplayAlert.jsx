import React from 'react';
import { Link } from 'react-router';

let DisplayAlert = React.createClass({
  render() {
    return (
      <div className="nothing-found">
          <div className="copy">
            {this.props.errorInfo.messages.map((message, index) => <div key={'msg_'+index}>{message}</div>)}
          </div>
          <div className="buttons">
            <Link to={this.props.tryUrl} className="big-button new-search-button" role="button">Try Again</Link>
          </div>
      </div>
    )
  }
});

DisplayAlert.defaultProps = {
  errorInfo: {
    type: 'Error.Search.Generic',
    messages: ["Your request cannot be processed", "at the moment due to technical problems.", "Please try again later"]
  },
  tryUrl: "/search"
};

export default DisplayAlert;
