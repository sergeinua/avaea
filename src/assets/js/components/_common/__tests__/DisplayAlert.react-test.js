import React from 'react';
import DisplayAlert from '../DisplayAlert';
import renderer from 'react-test-renderer';
import Link from 'react';

test('DisplayAlert component test', () => {

  var component = renderer.create(
    <DisplayAlert />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

});
