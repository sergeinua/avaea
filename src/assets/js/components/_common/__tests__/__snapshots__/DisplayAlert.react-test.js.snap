exports[`test DisplayAlert component test 1`] = `
<div
  className="nothing-found">
  <div
    className="copy">
    <div>
      Your request cannot be processed
    </div>
    <div>
      at the moment due to technical problems.
    </div>
    <div>
      Please try again later
    </div>
  </div>
  <div
    className="buttons">
    <a
      className="big-button new-search-button"
      onClick={[Function]}
      role="button"
      style={Object {}}>
      Try Again
    </a>
  </div>
</div>
`;
